### DISCUSSION TITLE
I can't figure out how to import a vue 3 component ?

### DISCUSSION THREAD

**Discussion by 3aluw** [nuxt/nuxt#22764] [Questions] [answered]
I am trying to use [vue3-apexcharts](https://github.com/apexcharts/vue3-apexcharts) in nuxt 3.
I followed this section about [vue-plugins](https://nuxt.com/docs/guide/directory-structure/plugins#vue-plugins) .
So I created a plugin and used this simple code :
```
import VueApexCharts from "vue3-apexcharts"
export default defineNuxtPlugin(nuxtApp => {
const apexchart = VueApexCharts;
nuxtApp.vueApp.use(apexchart)})
```
now, in the template :  `<apexchart>` gives me an error  :

> Failed to resolve component: apexchart
> If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.

I tried to import it :
const { apexchart } = useNuxtApp().vueApp;
I got this error :

>  Component <Anonymous> is missing template or render function.

So how can I do it ?

**Comment by Ok9xNirab** [comment, marked as answer]
## For Component

```js
nuxtApp.vueApp.component("apexchart", apexchart);
```

For example, I'm using [@vueform/slider](https://github.com/vueform/slider).

```js
import Slider from "@vueform/slider";
import "@vueform/slider/themes/default.css";

export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.vueApp.component("Slider", Slider);
});
```

If vue-3 package / component doesn't support SSR, You need to use plugin file name something like `<plugin-name>.client.ts`

and use client side render only:
```vue
<ClientOnly>
    <Slider
      :min="0"
      :max="1000"
      :tooltips="false"
      v-model="form.distance" />
</ClientOnly>

```

I hope you understand, how to use vue-3 component in nuxt 3 !

**Comment by 3aluw** [comment]
Thank you, I used .client naming but didn't know I need to wrap the components by`<ClientOnly>`.
However, I tried defining the plugin by `.use` and it is working. ` nuxtApp.vueApp.use(VueApexCharts); `
So is there a difference between` .use `and `.component` in this situation ?

**Reply by aryaadinulfadlan** [reply]
i think the most important thing here is, you need to wrap apexchart component with `<ClientOnly>` tag. because it doesn't support for SSR.
For the differences between `.use` and `.component` :
1. you can use both 
2. but if you use `.use` approach you will be using the default component's name (apexchart)
3. if you use `.component` approach, you will be able to customize the component's name => `nuxtApp.vueApp.component("MyApexcharts", VueApexCharts)`
